src/docfx/lib/json/JsonUtility.cs

```diff
@@ -238,7 +245,18 @@ public static (List<Error>, JToken) Parse(string json)
 
         public static void Merge(JObject container, JObject overwrite)
         {
-            container.Merge(overwrite, s_mergeSettings);
+            foreach (var (key, value) in overwrite)
+            {
+                if (container[key] is JObject containerObj && value is JObject overwriteObj)
+                {
+                    Merge(containerObj, overwriteObj);
+                }
+                else if (IsNullOrUndefined(container[key]) || !IsNullOrUndefined(value))
```

Comment:
how to handle JArray?

Reply:
It seems not trivial to rewrite `Merge`, there is a workaround to keep line info inside overwrite
https://github.com/dotnet/docfx/blob/611e336cb75493dad8a5979306148d8ccabdcdb9/src/docfx/config/ConfigLoader.cs#L167

Reply:
Add more tests to make sure the implementation could fulfill our needs